File: stripe/src/main/java/com/stripe/android/view/ErrorListener.java
package com.stripe.android.view;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TextInputLayout;

class ErrorListener implements StripeEditText.ErrorMessageListener {

    @NonNull private final TextInputLayout mTextInputLayout;

    ErrorListener(@NonNull TextInputLayout textInputLayout) {
        mTextInputLayout = textInputLayout;
    }

    @Override
    public void displayErrorMessage(@Nullable String message) {
        if (message == null) {
            mTextInputLayout.setErrorEnabled(false);
        } else {
            mTextInputLayout.setError(message);
        }
    }
}

File: stripe/src/main/java/com/stripe/android/view/ErrorListener.kt
package com.stripe.android.view

import com.google.android.material.textfield.TextInputLayout

internal class ErrorListener(
    private val textInputLayout: TextInputLayout
) : StripeEditText.ErrorMessageListener {

    override fun displayErrorMessage(message: String?) {
        if (message == null) {
            textInputLayout.error = null
            textInputLayout.isErrorEnabled = false
        } else {
            textInputLayout.error = message
        }
    }
}
